### PR TITLE
PIC-2292

### PR DESCRIPTION
**Pull request by vb-moj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/OffenceEntity.java [hunk 1]
```diff
@@ -60,4 +65,11 @@ public class OffenceEntity extends BaseImmutableEntity implements Serializable
 
     @Column(name = "LIST_NO", nullable = false)
     private final Integer listNo;
+
+    @LazyCollection(value = LazyCollectionOption.FALSE)
+    @JsonIgnore
+    @OneToMany(mappedBy = "offence", orphanRemoval=true, cascade = CascadeType.ALL)
+    @OrderColumn(name = "JUDICIAL_RESULTS_ORDER", nullable = false)
```
Comment (vb-moj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/OffenceEntity.java:72]: Testing the order column atm, By default hibernate should manage the list order using this column.

File: src/test/java/uk/gov/justice/probation/courtcaseservice/controller/model/CourtCaseRequestTest.java [hunk 2]
```diff
@@ -51,8 +53,8 @@ void givenAllFieldsPresent_whenCallAsEntity_returnHearingEntity() {
                 .breach(BREACH)
                 .preSentenceActivity(PRE_SENTENCE_ACTIVITY)
                 .offences(Arrays.asList(
-                                new OffenceRequestResponse("OFFENCE_TITLE1", "OFFENCE_SUMMARY1", "ACT1", 10),
-                                new OffenceRequestResponse("OFFENCE_TITLE2", "OFFENCE_SUMMARY2", "ACT2", 20)
+                                new OffenceRequestResponse("OFFENCE_TITLE1", "OFFENCE_SUMMARY1", "ACT1", 10, Collections.emptyList()),
+                                new OffenceRequestResponse("OFFENCE_TITLE2", "OFFENCE_SUMMARY2", "ACT2", 20, Collections.emptyList())
```
Comment (vb-moj) [inline src/test/java/uk/gov/justice/probation/courtcaseservice/controller/model/CourtCaseRequestTest.java:57]: Updating the tests ATM.

Comment (jhevans) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/controller/model/CourtCaseRequestTest.java:57]: Currently we don't have a concrete use case for this data other than for emitting events, I think we can leave it for now and add it when we need it. If we're interested we can get it from the extended endpoint which is the main thing.

**Review by jhevans — DISMISSED** [review]
LGTM 👍

**Comment by jhevans** [comment]
Approved but there are still failing tests

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/OffenceEntity.java (hunk 1 above)
Comment (vb-moj) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/OffenceEntity.java:72]: IT confirms that hibernate maintains the order of insert without any extra logic.

**Comment by vb-moj** [comment]
Closing this PR as some re work is needed.

**Pull request closed without merging**